tpu-client-next/src/connection_workers_scheduler.rs

```diff
@@ -217,6 +230,33 @@ impl ConnectionWorkersScheduler {
         cancel: CancellationToken,
     ) -> Result<Self, ConnectionWorkersSchedulerError> {
         let endpoint = Self::setup_endpoint(bind, stake_identity)?;
+
+        let connection_gen = Arc::new(AtomicU64::new(0));
```

Comment:
This atomic variable identifies generation of the certificates. When we create a new worker holding connection, it has generation which corresponds to the moment of creation. Later it is used to check vs the generation in this atomic to verify that worker is valid.

I think that the probability that while we are updating endpoint certificate someone sends transactions such that these transactions will be sent with old certificate but the new one already has arrived, is negligible.